src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+    public static final User USER_WITH_EMPTY_PASSWORD = new User("Lukas Rodriguez", "", 49);
+    public static final User USER_WITH_ZERO_AGE = new User("Clayton Young", "6S+8(mc4Ku", 0);
+    public static final User USER_AGE_18 = new User("Collin Hayes", "6S+8(mc4Ku", 18);
+    public static final User USER_AGE_19 = new User("Pierson Lee", "7eLj7Jv(@9", 19);
+    public static final User USER_AGE_17 = new User("Jake Barnes", "sjK457_sP@", 17);
+    public static final User USER_WITH_NEGATIVE_AGE = new User("Clayton Young", "6S+8(mc4Ku", -20);
+    public static final User CORRECT_USER = new User("Oliver Gray", "bGcM@r55~6", 20);
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(NULL_USER));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_NULL_LOGIN));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_NULL_PASSWORD));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_NULL_AGE));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_EMPTY_LOGIN));
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_EMPTY_PASSWORD));
+    }
+
+    @Test
+    void register_zeroAge_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_ZERO_AGE));
+    }
+
+    @Test
+    void register_login5Char_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_LOGIN_5_CHARACTERS));
+    }
+
+    @Test
+    void register_password5Char_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_PASSWORD_5_CHARACTERS));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(USER_WITH_NEGATIVE_AGE));
+    }
+
+    @Test
+    void register_age17_notOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(USER_AGE_17));
+    }
+
+    @Test
+    void register_addExistUser_notOk() {
+        registrationService.register(CORRECT_USER);
```

Comment:
for such checks, you do not need to use your method, but you need to directly add it to storage, the checklist says about it